src/modules/scopes.js

```diff
@@ -72,63 +72,74 @@ function decorateRootScope($delegate, $parse) {
 
-  // var _watch = scopePrototype.$watch;
-  // scopePrototype.$watch = function (watchExpression, reactionFunction) {
-  //   var watchStr = humanReadableWatchExpression(watchExpression);
-  //   var scopeId = this.$id;
-  //   if (typeof watchExpression === 'function') {
-  //     arguments[0] = function () {
-  //       var start = perf.now();
-  //       var ret = watchExpression.apply(this, arguments);
-  //       var end = perf.now();
-  //       hint.emit('scope:watch', {
-  //         id: scopeId,
-  //         watch: watchStr,
-  //         time: end - start
-  //       });
-  //       return ret;
-  //     };
-  //   } else {
-  //     var thatScope = this;
-  //     arguments[0] = function () {
-  //       var start = perf.now();
-  //       var ret = thatScope.$eval(watchExpression);
-  //       var end = perf.now();
-  //       hint.emit('scope:watch', {
-  //         id: scopeId,
-  //         watch: watchStr,
-  //         time: end - start
-  //       });
-  //       return ret;
-  //     };
-  //   }
-
-  //   if (typeof reactionFunction === 'function') {
-  //     var applyStr = reactionFunction.toString();
-  //     arguments[1] = function () {
-  //       var start = perf.now();
-  //       var ret = reactionFunction.apply(this, arguments);
-  //       var end = perf.now();
-  //       hint.emit('scope:reaction', {
-  //         id: this.$id,
-  //         watch: watchStr,
-  //         time: end - start
-  //       });
-  //       return ret;
-  //     };
-  //   }
-
-  //   return _watch.apply(this, arguments);
-  // };
+  var _watch = scopePrototype.$watch;
+  var _digestEvents = [];
+  scopePrototype.$watch = function (watchExpression, reactionFunction) {
+    var watchStr = humanReadableWatchExpression(watchExpression);
+    var scopeId = this.$id;
+    if (typeof watchExpression === 'function') {
+      arguments[0] = function () {
+        var start = perf.now();
+        var ret = watchExpression.apply(this, arguments);
+        var end = perf.now();
+        _digestEvents.push({
+          eventType: 'scope:watch',
+          id: scopeId,
+          watch: watchStr,
+          time: end - start
+        });
+        return ret;
+      };
+    } else {
+      var thatScope = this;
+      arguments[0] = function () {
+        var start = perf.now();
+        var ret = thatScope.$eval(watchExpression);
+        var end = perf.now();
+        _digestEvents.push({
+          eventType: 'scope:watch',
+          id: scopeId,
+          watch: watchStr,
+          time: end - start
+        });
+        return ret;
+      };
+    }
+
+    if (typeof reactionFunction === 'function') {
+      arguments[1] = function () {
+        var start = perf.now();
+        var ret = reactionFunction.apply(this, arguments);
+        var end = perf.now();
+        _digestEvents.push({
+          eventType: 'scope:reaction',
+          id: this.$id,
+          watch: watchStr,
+          time: end - start
+        });
+        return ret;
```

Comment:
There's a few tweaks needed in your code (`if` statement and such) but I think that having clear intent trumps having fewer lines of code.  With the code you've written above, readers will need to look in two places to figure out what a particular action does (as opposed to now, where they only need one).